exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginsHandlerService.java

```diff
@@ -0,0 +1,130 @@
+
+/**
+ * Drill plugins handler, which allows to update storage plugins configs from the
+ * {@link CommonConstants#STORAGE_PLUGINS_OVERRIDE_CONF} conf file
+ *
+ * TODO: DRILL-6564: It can be improved with configs versioning and service of creating
+ * {@link CommonConstants#STORAGE_PLUGINS_OVERRIDE_CONF}
+ */
+public class StoragePluginsHandlerService implements StoragePluginsHandler {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(StoragePluginsHandlerService.class);
+
+  private final LogicalPlanPersistence hoconLogicalPlanPersistence;
+
+  public StoragePluginsHandlerService(DrillbitContext context) {
+    hoconLogicalPlanPersistence = new LogicalPlanPersistence(context.getConfig(), context.getClasspathScan(),
+        new ObjectMapper(new HoconFactory()));
+  }
+
+  @Override
+  public void loadPlugins(@NotNull PersistentStore<StoragePluginConfig> persistentStore,
+                          @Nullable StoragePlugins bootstrapPlugins) {
+    // if bootstrapPlugins is not null -- fresh Drill set up
+    StoragePlugins pluginsToBeWrittenToPersistentStore;
+
+    StoragePlugins newPlugins = getNewStoragePlugins();
+
+    if (newPlugins != null) {
+      pluginsToBeWrittenToPersistentStore = new StoragePlugins(new HashMap<>());
+      Optional.ofNullable(bootstrapPlugins)
+          .ifPresent(pluginsToBeWrittenToPersistentStore::putAll);
+
+      for (Map.Entry<String, StoragePluginConfig> newPlugin : newPlugins) {
+        String pluginName = newPlugin.getKey();
+        StoragePluginConfig oldPluginConfig = Optional.ofNullable(bootstrapPlugins)
+            .map(plugins -> plugins.getConfig(pluginName))
+            .orElse(persistentStore.get(pluginName));
+        StoragePluginConfig updatedStatusPluginConfig = updatePluginStatus(oldPluginConfig, newPlugin.getValue());
+        pluginsToBeWrittenToPersistentStore.put(pluginName, updatedStatusPluginConfig);
+      }
+    } else {
+      pluginsToBeWrittenToPersistentStore = bootstrapPlugins;
+    }
+
+    // load pluginsToBeWrittenToPersistentStore to Persistent Store
+    Optional.ofNullable(pluginsToBeWrittenToPersistentStore)
+        .ifPresent(plugins -> plugins.forEach(plugin -> persistentStore.put(plugin.getKey(), plugin.getValue())));
+  }
+
+  /**
+   * Helper method to identify the enabled status for new storage plugins config. If this status is absent in the updater
+   * file, the status is kept from the configs, which are going to be updated
+   *
+   * @param oldPluginConfig current storage plugin config from Persistent Store or bootstrap config file
+   * @param newPluginConfig new storage plugin config
+   * @return new storage plugin config with updated enabled status
+   */
+  private StoragePluginConfig updatePluginStatus(@Nullable StoragePluginConfig oldPluginConfig,
+                                                 @NotNull StoragePluginConfig newPluginConfig) {
+    if (!newPluginConfig.isEnabledStatusPresent()) {
```

Comment:
Am i missing anything here ? It looks like the assumption is that newPluginConfig will have all the keys that are present in oldPluginConfigs as well  which might not be the case. Since we are only updating the enabled key in this function. Whereas expectation is use the value from newPluginConfig for keys present in both oldPluginConfig and newPluginConfig and for keys only in oldPluginConfig preserve the old value.